docs/faq.rst

```diff
@@ -0,0 +1,106 @@
+.. _FAQ:
+
+FAQ
+===
+
+.. contents:: Question Lists
+  :local:
+
+Why does the warning related to ``num_communication_per_step`` or ``backward_passes_per_step`` pop up in Bluefog optimizer?
+--------------------------------------------------------------------------------------------
+
+During your usage of Bluefog optimizer, you may encounter the following the two types of warnings.
+
+
+.. code-block:: python
+
+   Warning (unexpected behavior):
+     After num_step_per_communication times of forward computation `y=model(x)` are called,
+     an optimizer step() function must be called.
+     It does not matter how many step() functions are called in between.
+     Please adjust num_step_per_communication to update model parameters locally.
+     More information can be found in the FAQ page.
+.. code-block:: python
+   
+   Warning (unexpected behavior):
+     After backward_passes_per_step times of backward computation `loss.backward()` are called,
+     an optimizer step() function must be called.
+     It does not matter how many step() functions are called in between.
+     Please adjust backward_passes_per_step to accumulate gradients locally.
+     More information can be found in the FAQ page.
+
+.. note::
+   The second warning is only encountered when using ``DistributedGradientAllreduceOptimizer`` or
+   ``DistributedAdaptThenCombineOptimizer``. The difference with other Bluefog optimizer is when
+   counting is triggered. All other Bluefog optimizers triggers this counting
+   behavior during forward computation, while ``DistributedGradientAllreduceOptimizer`` and 
+   ``DistributedAdaptThenCombineOptimizer`` triggers during backward computation.
+   This is also reflected by the optimizer argument naming.
+   All other optimizers uses ``num_step_per_communication``, while these two optimizers uses
+   ``backward_passes_per_step``.
+   The following discussion only focuses on forward computation case, but it works for backward
+   scenario as well.
+
+Consider the following admissible code snippet.
```

Comment:
+ To understand the meaning of the above two warnings, consider ..

Reply:
Done.